tests/tools/plugins/test_hooks.cc

```diff
@@ -288,6 +325,22 @@ TSPluginInit(int argc, const char *argv[])
   TSHttpHookAdd(TS_HTTP_SSN_CLOSE_HOOK, gCont);
   TSHttpHookAdd(TS_HTTP_TXN_START_HOOK, gCont);
   TSHttpHookAdd(TS_HTTP_TXN_CLOSE_HOOK, gCont);
+  TSHttpHookAdd(TS_SSL_CERT_HOOK, gCont);
+  TSHttpHookAdd(TS_SSL_SERVERNAME_HOOK, gCont);
+
+  // NOTE: as of January 2019 these two hooks are only triggered for TLS connections.  It seems that, at trafficserver
+  // startup, spurious data on the TLS TCP port may cause trafficserver to attempt (and fail) to create a TLS
+  // connection.  If this happens, it will result in TS_VCONN_START_HOOK being triggered, and then TS_VCONN_CLOSE_HOOK
+  // will be triggered when the connection closes due to failure.
+  //
+  TSHttpHookAdd(TS_VCONN_START_HOOK, gCont);
+  TSHttpHookAdd(TS_VCONN_CLOSE_HOOK, gCont);
+
+  // TSHttpHookAdd(TS_SSL_SESSION_HOOK, gCont); -- Event is TS_EVENT_SSL_SESSION_NEW -- Event data is TSHttpSsn
+  // TSHttpHookAdd(TS_SSL_SERVER_VERIFY_HOOK, gCont);
```

Comment:
Is there a reason these are commented out?  Why are the hooks not being called?

Reply:
They don't seem to be working.